agent/Cargo.toml

```
@@ -35,6 +35,7 @@ serde = "1.0.104"
 serde_derive = "1.0.104"
 serde_json = "1.0.45"
 serde_yaml = { version = "0.8.11", optional = true }
+simple-mermaid = { version = "0.1"} # used for docs
```

seems a bit strange to add a dependency to the package just for documentation. I wish there was a `docs-dependencies` section like there is `dev-dependencies`.
```suggestion
simple-mermaid = "0.1" # used for docs
```